lib/vmdb/util.rb

```diff
@@ -1,7 +1,10 @@
 module VMDB
   module Util
-    def self.http_proxy_uri
-      proxy = VMDB::Config.new("vmdb").config[:http_proxy] || {}
+    def self.http_proxy_uri(proxy_config = :default)
+      # TODO: (julian) This looks really messy. Need to make it look nicer.
+      vmdb_proxy = VMDB::Config.new("vmdb").config[:http_proxy]
+      return nil unless vmdb_proxy
+      proxy = vmdb_proxy[proxy_config] || {}
```

Comment:
So, it looks like if this were on an upgraded appliance with "legacy" configs (i.e., no `:default` under `:http_proxy`, just the proxy settings themselves) then the worst case is that the proxy settings would be empty.

I think you should log a warning here saying that you couldn't find any proxy settings for the given `proxy_config`.  That way, legacy files will at least warn the user that they need to update their proxy configs.